internal/cmd/gentypes/main.go

```diff
@@ -615,3 +669,32 @@ func rename(from, to string) patch {
 		return fmt.Errorf("no member named %q", from)
 	}
 }
+
+// replaceUnionWithBytes replaces the union with bytes within a struct.
+// you can specify the union name or the union position (anonymous).
+// the position can be -1, it means the first union in struct will be replaced
+// otherwise with a given position or name, it replaces exactly the specified
+// union name or union position with bytes.
+func replaceUnionWithBytes(name string, position int, bytesName string) patch {
```

Comment:
I think you can decompose this into two steps instead: `name(3, "extra"), replaceUnionWithBytes("extra")`. See the discussion re `.Extra` though, I think the approach won't work.

```
func name(member int, name string) patch {
	return modifyNth(func(m *btf.Member) error {
		if m.Name != "" {
			return fmt.Errorf("member already has name %q", m.Name)
		}

		m.Name = name
		return nil
	}, member)
}
```

Reply:
Decomposed.